src/cecs277TermProject/Logger.java

```diff
@@ -34,15 +37,19 @@ private Logger() throws FileNotFoundException
 	{
 		this.filename = "roach_motel_log.txt";
 		this.outFile = new File(filename);
+		PrintWriter out = null;
 		
 	}
 	
 	public void log(String message)
 	{
 		try 
 		{
-			PrintWriter out = new PrintWriter(outFile);
-			out.println(message);
+			Instant time = Instant.now();
+			DateTimeFormatter format = DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss:SSS");
+			String timestamp = format.format(time);
+			out = new PrintWriter(outFile);
```

Comment:
Move this to the constructor because this method will be called a lot and we don't want to create a new PrintWriter object everytime.